// react-native-reanimated-table/lib/module/components/cols.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Cell } from './cell';
import { sum } from '../utils';
export const Col = _ref => {
  let {
    data,
    style,
    width,
    heightArr,
    flex,
    textStyle,
    ...props
  } = _ref;
  return data ? /*#__PURE__*/React.createElement(View, {
    style: StyleSheet.flatten([{
      width: width ?? (flex ? undefined : 1),
      flex
    }, style])
  }, data.map((item, i) => {
    const height = heightArr === null || heightArr === void 0 ? void 0 : heightArr[i];
    return /*#__PURE__*/React.createElement(Cell, _extends({
      key: i,
      data: item,
      width: width,
      height: height,
      textStyle: textStyle
    }, props));
  })) : null;
};
export const Cols = _ref2 => {
  let {
    data,
    style,
    widthArr,
    heightArr,
    flexArr,
    textStyle,
    ...props
  } = _ref2;
  const width = widthArr ? sum(widthArr) : 0;
  return data ? /*#__PURE__*/React.createElement(View, {
    style: StyleSheet.flatten([styles.cols, width ? {
      width
    } : {}])
  }, data.map((item, i) => {
    const flex = flexArr === null || flexArr === void 0 ? void 0 : flexArr[i];
    const wth = widthArr === null || widthArr === void 0 ? void 0 : widthArr[i];
    return /*#__PURE__*/React.createElement(Col, _extends({
      key: i,
      data: item,
      width: wth,
      heightArr: heightArr,
      flex: flex,
      style: style,
      textStyle: textStyle
    }, props));
  })) : null;
};
const styles = StyleSheet.create({
  cols: {
    flexDirection: 'row'
  }
});
